numpyro/infer/util.py

```diff
@@ -427,6 +427,21 @@ def initialize_model(rng_key, model,
 
     if not_jax_tracer(is_valid):
         if device_get(~jnp.all(is_valid)):
+            with numpyro.validation_enabled(), trace() as tr:
+                # validate parameters
+                substituted_model(*model_args, **model_kwargs)
+                # validate values
+                for site in tr.values():
+                    if site['type'] == 'sample':
+                        with warnings.catch_warnings(record=True) as ws:
+                            site['fn']._validate_sample(site['value'])
+                        if len(ws) > 0:
+                            for w in ws:
+                                # at site information to the warning message
+                                w.message.args = ("Site {}: {}".format(site["name"], w.message.args[0]),) \
+                                    + w.message.args[1:]
```

Comment:
What does a sample warning message look like?

Reply:
Thanks for reviewing, @neerajprad! I'm a bit busy today but will run this again to display the warning message tomorrow. Here I just want to add site information to the warning message, because not many users know how to use `warnings` to turn a warning to an error to debug.

Reply:
@neerajprad the following model
```python
import numpyro
import numpy as np
def model():
    x = numpyro.sample("x", numpyro.distributions.Normal())
    numpyro.sample("obs", numpyro.distributions.Normal(x), obs=float('nan'))
mcmc = numpyro.infer.MCMC(numpyro.infer.NUTS(model), 10, 10)
mcmc.run(np.array([0, 0], dtype='uint32'))
```
gives the warning
```
UserWarning: Site obs: Out-of-support values provided to log prob method. The value argument should be within the support.
```